service/storage_service.cc

```diff
@@ -728,9 +733,13 @@ class storage_service::raft_ip_address_updater: public gms::i_endpoint_state_cha
             (void)_ss._group0->client().hold_read_apply_mutex().then([this, id, endpoint, prev_ip](semaphore_units<> g) {
                 const auto hid = locator::host_id{id.uuid()};
                 if (_address_map.find(id) == endpoint && _ss.get_token_metadata().get_endpoint_for_host_id_if_known(hid) != endpoint) {
-                    return _ss.mutate_token_metadata([this, hid, prev_ip](mutable_token_metadata_ptr t) {
-                        return _ss.sync_raft_topology_nodes(std::move(t), hid, prev_ip, {});
-                    }).finally([g = std::move(g)]{});
+                    return utils::get_local_injector().inject("ip-change-raft-sync-delay", std::chrono::milliseconds(500))
+                        .then([this, hid, prev_ip]() {
```

Comment:
I see we're piling more and more `then`s, perhaps it's time to coroutinize this code?